release-23.1: roachprod: automatically create admin user for secure clusters

renatolabs opened the pull request:
Backport 2/2 commits from #112765, 1/1commits from #112985.

/cc @cockroachdb/release

Release justification: test-only changes.

---

This commit adds a default admin user to clusters created with roachprod; this applies both to the system interface, as well as virtual clusters (shared or separate process).

This is mostly done for convenience, so that roachprod users can have easy access to the dbconsole. It is also helpful while running roachtests.

Epic: none

Release note: None

Commit:
roachprod: automatically create admin user for secure clusters

This commit adds a default admin user to clusters created with
roachprod; this applies both to the system interface, as well as
virtual clusters (shared or separate process).

This is mostly done for convenience, so that roachprod users can have
easy access to the dbconsole. It is also helpful while running
roachtests.

Epic: none

Release note: None

Commit:
roachprod: support `--secure` mode in `stop-sql`

Previously, shared-process virtual clusters serviced by secure
clusters would not be able to be stopped using `stop-sql`. This was
made worse by the fact that the error checking was wrong, and the
error returned by the `cockroach sql` call was being ignored.

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/112973)
<!-- Reviewable:end -->

Commit:
roachtest: ignore errors if default admin user cannot be created

In #112765, we started creating default admin users for secure
clusters created with roachprod. The new function would return an
error if the user  could not be created.

Unfortunately, that behaviour is not achievable: roachprod is used in
a variety of different contexts, especially within roachtests, and a
failure to create a user might be "expected" behaviour depending on
what the test is doing.

In this commit, we just log failures to create admin users.

Fixes: #112943.

Release note: None
